### PR TITLE
[DRFT-310] Allows fetch_url to accept HTTP method parameter

### PR DESCRIPTION
**Pull request by petrblaho**
## Secure Coding Practices Checklist Link
- https://github.com/RedHatInsights/secure-coding-checklist

## Secure Coding Checklist
- [ ] Input Validation
- [ ] Output Encoding
- [ ] Authentication and Password Management
- [ ] Session Management
- [ ] Access Control
- [ ] Cryptographic Practices
- [x] Error Handling and Logging
- [ ] Data Protection
- [ ] Communication Security
- [ ] System Configuration
- [ ] Database Security
- [ ] File Management
- [ ] Memory Management
- [x] General Coding Practices

### CODE REVIEW

File: kerlescan/exceptions.py [hunk 1]
```diff
@@ -53,3 +53,12 @@ def __init__(self, message):
...
         super(RuntimeError, self).__init__()
         self.message = message
+
+
+class IllegalHttpMethodError(ValueError):
+    def __init__(self, message):
+        """
+        Raise this exception if we detect illegal HTTP method
+        """
+        super(IllegalHttpMethodError, self).__init__()
```
Comment (vkrizan) [inline kerlescan/exceptions.py:63]: Nit: In Python 2.7+ you can call `super().__init__()`

Also if the message isn't passed to `ValueError`, then calling `str()` would give an empty string.

File: kerlescan/service_interface.py [hunk 2]
```diff
@@ -48,14 +53,20 @@ def _validate_service_response(response, logger, auth_header):
         raise ServiceError("Error received from backend service")
 
 
-def fetch_url(url, auth_header, logger, time_metric, exception_metric):
+def fetch_url(url, auth_header, logger, time_metric, exception_metric, method="get"):
     """
     helper to make a single request
     """
+
+    if method not in ["get", "options", "head", "post", "put", "patch", "delete"]:
```
Comment (vkrizan) [inline kerlescan/service_interface.py:61]: Observation: The list can be a constant.

**Review by vkrizan — APPROVED** [review]
LGTM. Thanks.

File: kerlescan/exceptions.py (hunk 1 above)
Comment (petrblaho) [reply kerlescan/exceptions.py]: I am aware of that `str()` thing.

Thanks for that `super()` notation.

**Pull request merged**